src/XMakeBuildEngine/Evaluation/ProjectParser.cs

```diff
@@ -358,6 +356,27 @@ private ProjectItemElement ParseProjectItemElement(XmlElementWithLocation elemen
 
             ProjectItemElement item = new ProjectItemElement(element, parent, _project);
 
+            foreach (XmlAttributeWithLocation attribute in element.Attributes)
+            {
+                bool isKnownAttribute;
+                bool isValidMetadataNameInAttribute;
+
+                CheckMetadataAsAttributeName(attribute.Name, out isKnownAttribute, out isValidMetadataNameInAttribute);
```

Comment:
Should this invalid attribute checking logic also be callable from the construction APIs? Otherwise API callers might set an invalid metadata name and only know about it later when the xml does not parse.